aws_topology/stackstate_checks/aws_topology/resources/kinesis.py

```
@@ -1,88 +1,118 @@
-    CLOUDFORMATION_TYPE = 'AWS::Kinesis::Stream'
+    CLOUDFORMATION_TYPE = "AWS::Kinesis::Stream"
+
+    @set_required_access_v2("kinesis:ListTagsForStream")
+    def collect_tags(self, stream_name):
+        self.client.list_tags_for_stream(StreamName=stream_name).get("Tags", [])
+
+    @set_required_access_v2("kinesis:DescribeStreamSummary")
+    def collect_stream_description(self, stream_name):
+        return self.client.describe_stream_summary(StreamName=stream_name)
+
+    def collect_stream(self, stream_name):
+        data = self.collect_stream_description(stream_name)
+        tags = self.collect_tags(stream_name)
+        return StreamData(stream=data, tags=tags)
+
+    @set_required_access_v2("kinesis:ListStreams")
+    def collect_streams(self):
+        for stream in [
+            self.collect_stream(stream_name)
+            for stream_name in client_array_operation(self.client, "list_streams", "StreamNames")
+        ]:
+            yield stream
 
     def process_all(self, filter=None):
-        for list_streams_page in self.client.get_paginator('list_streams').paginate():
-            for stream_name in list_streams_page.get('StreamNames') or []:
-                self.process_stream(stream_name)
-
-    def process_stream(self, stream_name):
-        stream_summary_raw = self.client.describe_stream_summary(StreamName=stream_name)
-        stream_summary = make_valid_data(stream_summary_raw)
-        stream_tags = self.client.list_tags_for_stream(StreamName=stream_name).get('Tags') or []
-        stream_summary['Name'] = stream_name
-        stream_summary['Tags'] = stream_tags
-        stream_arn = stream_summary['StreamDescriptionSummary']['StreamARN']
-        self.emit_component(stream_arn, self.COMPONENT_TYPE, stream_summary)
+        if not filter or "kinesis" in filter:
+            for stream_data in self.collect_streams():
+                try:
+                    self.process_stream(stream_data)
+                except Exception:
```

Basically we shoud never use catch all exceptions, think it can be removed
There are unrecoverable exceptions like "out of memory"
Or "system halts"